sgkit/tests/test_popgen.py

```diff
@@ -37,39 +40,70 @@ def ts_to_dataset(ts, samples=None):
 def test_diversity(size):
```

Comment:
Given that map_blocks is underlying these stats, I think chunking on variants and samples dimensions would be useful parameters for these tests.  I'm imagining something like:

```python
@pytest.mark.parametrize("size", [2, 3, 10, 100])
@pytest.mark.parametrize("chunks", [(-1, -1), (10, 10)])
def test_diversity(size, chunks):
    ts = msprime.simulate(size, length=100, mutation_rate=0.05, random_seed=42)
    ds = ts_to_dataset(ts).chunk(dict(zip(["samples", "variants"], chunks)))
```

Reply:
Good call. Adding this test uncovered two bugs: one to do with the wrong `numblocks` index being used, and another where the `guvectorize` function if the input is chunked along the samples dimension. I haven't been able to figure out how fix the latter yet, so I'm going to leave it for the moment. (Popgen doesn't need to chunk along the samples dimension just yet.)

Reply:
I've added warnings to the documentation about not supporting chunking in the samples dimension.